app/src/main/java/com/example/android/sunshine/app/LocationEditTextPreference.java

```diff
@@ -52,6 +62,34 @@ private void init(Context context, AttributeSet attrs) {
                 a.recycle();
             }
         }
+
+        GoogleApiAvailability apiAvailability = GoogleApiAvailability.getInstance();
+        int resultCode = apiAvailability.isGooglePlayServicesAvailable(context);
+        if (resultCode == ConnectionResult.SUCCESS) {
+            // Add the get current location widget to our location preference
+            setWidgetLayoutResource(R.layout.pref_current_location);
+        }
+    }
+
+    @Override
+    protected View onCreateView(ViewGroup parent) {
+        View view = super.onCreateView(parent);
+        View currentLocation = view.findViewById(R.id.current_location);
+        currentLocation.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                PlacePicker.IntentBuilder intentBuilder = new PlacePicker.IntentBuilder();
+
+                try {
+                    Activity activity = (Activity) getContext();
+                    Intent intent = intentBuilder.build(activity);
+                    activity.startActivityForResult(intent, SettingsActivity.PLACE_PICKER_REQUEST);
```

Comment:
sure this would be safer if done from the activity? IMO it's always more organised if you keep the navigation always inside the activity - so you'd just callback to the activity here and do the navigation from there (specially because it's starting for result)